### PR TITLE
[RFC] posix: signal: empty + fill + set

### LINKED ISSUE #59927: posix: implement `sigaddset()`
There are currently 20 missing POSIX functions with trivial implementations, `sigaddset()` is one of them.

> sigaddset - add a signal to a signal set
> 
> The sigaddset() function adds the individual signal specified by the signo to the signal set pointed to by set.
> 
> Applications shall call either [sigemptyset()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigemptyset.html) or [sigfillset()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigfillset.html) at least once for each object of type sigset_t prior to any other use of that object. If such an object is not initialized in this way, but is nonetheless supplied as an argument to any of [pthread_sigmask()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_sigmask.html), [sigaction()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigaction.html), sigaddset(), [sigdelset()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigdelset.html), [sigismember()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigismember.html), [sigpending()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigpending.html), [sigprocmask()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigprocmask.html), [sigsuspend()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigsuspend.html), [sigtimedwait()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigtimedwait.html), [sigwait()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigwait.html), or [sigwaitinfo()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigwaitinfo.html), the results are undefined.

See https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigaddset.html

See RFC #51211 for more info.

### LINKED ISSUE #59929: posix: implement `sigemptyset()`
There are currently 20 missing POSIX functions with trivial implementations, `sigemptyset()` is one of them.

> sigemptyset - initialize and empty a signal set
> 
> The sigemptyset() function initializes the signal set pointed to by set, such that all signals defined in POSIX.1-2017 are excluded.

See https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigemptyset.html

See RFC #51211 for more info.

### LINKED ISSUE #59931: posix: implement `sigfillset()`
There are currently 20 missing POSIX functions with trivial implementations, `sigfillset()` is one of them.

> sigfillset - initialize and fill a signal set
> 
> The sigfillset() function shall initialize the signal set pointed to by set, such that all signals defined in this volume of POSIX.1-2017 are included.

See https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigfillset.html

See RFC #51211 for more info.

### PR DESCRIPTION
**Pull request by ycsin**
Implementation for `sigemptyset`, `sigfillset`, `sigaddset`
Heavily 'influenced' by nuttx's implementation

will probably finish off the remaining signal APIs tomorrow..

fixes #59927
fixes #59929
fixes #59931

### CODE REVIEW

File: include/zephyr/posix/signal.h [hunk 1]
```diff
@@ -12,6 +12,60 @@
...
+	SIGSYS,
+	SIGTRAP,
+	SIGURG,
+	SIGVTALRM,
+	SIGXCPU,
+	SIGXFSZ,
+};
+
+/* Signal set management definitions and macros. */
+#define MIN_SIGNO       1                       /* Lowest valid signal number */
```
Comment (ycsin) [inline include/zephyr/posix/signal.h:47]: I might move some of these into the enum above

Comment (keith-packard) [reply include/zephyr/posix/signal.h:47]: I'm somewhat concerned that using an ENUM will allow pre-processor defines to override these without generating a warning.

Comment (ycsin) [reply include/zephyr/posix/signal.h:48]: Maybe I shouldn't be lazy and should have used defines instead..

Comment (cfriedt) [reply include/zephyr/posix/signal.h:48]: I think technically you can do both 😁 but some people might find that a bit too verbose

Comment (cfriedt) [reply include/zephyr/posix/signal.h:48]: Generally though, `#define`s would give you more compile-time constness which is A Good Thing™️ for size.

File: include/zephyr/posix/signal.h [hunk 2]
```diff
@@ -12,6 +12,62 @@
...
+
+/* Definitions for "standard" signals */
+#define SIGSTDMIN       1                           /* First standard signal number */
+#define SIGSTDMAX       CONFIG_POSIX_SIGNAL_STD_MAX /* Last standard signal number */
+
+/* Definitions for "real time" signals */
+#define SIGRTMIN        (SIGSTDMAX + 1) /* First real time signal */
+#define SIGRTMAX        MAX_SIGNO       /* Last real time signal */
+#define _NSIG           (MAX_SIGNO + 1) /* Biggest signal number + 1 */
+#define NSIG            _NSIG           /* _NSIG variant commonly used */
```
Comment (cfriedt) [inline include/zephyr/posix/signal.h:59]: are `_NSIG`, `NSIG`, `MAX_SIGNO`, `SIGSTDMAX`, `SIGSTDMIN`, etc standard names? I haven't looked yet.

Comment (ycsin) [reply include/zephyr/posix/signal.h:59]: > `_NSIG`, `NSIG`

I can't find this in the standard, but pretty common in implementations to have these for calculation:

https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/signal.h#L7

File: include/zephyr/posix/signal.h [hunk 3]
```diff
@@ -12,6 +12,62 @@
...
+
+/* sigset_t is represented as an array of 32-b unsigned integers.
+ * _SIGSET_NELEM is the allocated size of the array
+ */
+#define _SIGSET_NELEM   ((_NSIG + 31) >> 5)
```
[the remark this answers is not in the document]
Comment (ycsin) [reply include/zephyr/posix/signal.h:64]: Not really standard, different implementations seems to have different names for this, to calculate the `uint32_t` array size to fit the number of signals, in nuttx its `_SIGSET_NELEM`, in Linux its `_NSIG_WORDS`

**Comment by ycsin** [comment]
closed, see #60083

**Pull request closed without merging**